lib/neo4j/shared/permitted_attributes.rb

```diff
@@ -0,0 +1,23 @@
+module Neo4j::Shared
+  module PermittedAttributes
+    extend ActiveSupport::Concern
+    include ActiveModel::ForbiddenAttributesProtection
+
+    def process_attributes(attributes)
+      attributes = sanitize_for_mass_assignment(attributes)
+      super(attributes)
+    end
+
+    def attributes=(attributes)
+      attributes = sanitize_for_mass_assignment(attributes)
+      super(attributes)
+    end
+
+    protected
+
+    def sanitize_for_mass_assignment(attributes)
+      attributes = super(attributes)
+      attributes.respond_to?(:symbolize_keys) ? attributes.symbolize_keys : attributes
```

Comment:
@cheerfulstoic  I moved it out to allow `ActiveRel` to be supported as well but... I haven't tested it. I should, indeed.

About symbolize_keys, maybe it's better to split it as you said. ;)